### PR TITLE
Calibration refactor

### PR DESCRIPTION
**Pull request by julianoes**
This is a first stab at refactoring how the sensor calibration values are brought to the sensor drivers. For Snapdragon/QURT using the DriverFramework ioctl calls are not available and therefore uORB messages are used.

~~Note to self: the df_hmc5883 driver is not subscribing to the calbration topic yet.~~

@LorenzMeier Is this somewhat close to what you envisioned?

### CODE REVIEW

**Comment by LorenzMeier** [comment]
This is looking excellent. All what is needed now is to run an orb_check() in the sensor loop and copy the calibration data (my apologies if I missed that) and this should be good to go to hit master after testing.

**Comment by julianoes** [comment]
@LorenzMeier how do you mean orb_check() in the sensor loop? Do you mean in sensors.cpp, or in the respective drivers?

**Comment by LorenzMeier** [comment]
In each driver. Best rate-limit it to 10 Hz or so (using a _last_cal_check variable and the hrt_absolute_time())

**Comment by julianoes** [comment]
Ok, "in each driver" is done. Do you think thath rate limiting using hrt_absolute_time() is cheaper than orb_check every time?

**Comment by LorenzMeier** [comment]
Not sure about which one is more efficient - go with what is faster to implement.

**Comment by julianoes** [comment]
This state seems to work on the Snapdragon and was also regression tested on a Pixhawk.

File: src/modules/commander/accelerometer_calibration.cpp [hunk 1]
```diff
@@ -181,22 +180,21 @@ int do_accel_calibration(int mavlink_fd)
...
 	int res = OK;
 
 	char str[30];
 
-#ifndef __PX4_QURT
 	/* reset all sensors */
 	for (unsigned s = 0; s < max_accel_sens; s++) {
+#ifndef __PX4_QURT
 		sprintf(str, "%s%u", ACCEL_BASE_DEVICE_PATH, s);
 		/* reset all offsets to zero and all scales to one */
```
Comment (LorenzMeier) [inline src/modules/commander/accelerometer_calibration.cpp]: If you are not resetting the offsets here, how do you ensure that the resulting calibration did you factor in the previously calibrated offsets? We can agree to remove this, but if yes, then we need to make sure to calibrate off a measurement which does not contain the correction from the offsets.

**Comment by LorenzMeier** [comment]
It looks overall good, but on non-NuttX devices doing two subsequent calibrations will give you wrong results. I do have an idea how to change things so it becomes easier and it works.

File: src/modules/commander/accelerometer_calibration.cpp (hunk 1 above)
Comment (julianoes) [inline src/modules/commander/accelerometer_calibration.cpp]: On QURT, we do reset all the params, just not the ioctl. The use case that you describe works when I tested it.

**Comment by julianoes** [comment]
@LorenzMeier I've tested all calibrations with Pixhawk and did a quick hover in SITL. I consider this good to be merged.

**Pull request merged**